core/src/main/java/hudson/FilePath.java

```diff
@@ -482,21 +483,39 @@ public int zip(OutputStream out, DirScanner scanner) throws IOException, Interru
      *      is archived.
      */
     public int archive(final ArchiverFactory factory, OutputStream os, final DirScanner scanner) throws IOException, InterruptedException {
+        Consumer<String> c = (Consumer<String> & Serializable)(s) -> {};
+        return archive(factory, os, scanner, c);
+    }
+
+    /**
+     * Archives this directory into the specified archive format, to the given {@link OutputStream}, by using
+     * {@link DirScanner} to choose what files to include.
+     *
+     * a Consumer {@link Consumer} is passed to process the list of stashed files
+     *
+     * @return
+     *      number of files/directories archived. This is only really useful to check for a situation where nothing
+     *      is archived.
+     */
+    public int archive(final ArchiverFactory factory, OutputStream os, final DirScanner scanner, Consumer<String> f) throws IOException, InterruptedException {
         final OutputStream out = (channel!=null)?new RemoteOutputStream(os):os;
-        return act(new Archive(factory, out, scanner));
+        return act(new Archive(factory, out, scanner, f));
     }
     private class Archive extends SecureFileCallable<Integer> {
         private final ArchiverFactory factory;
         private final OutputStream out;
         private final DirScanner scanner;
-        Archive(ArchiverFactory factory, OutputStream out, DirScanner scanner) {
+        private final Consumer<String> function;
```

Comment:
could you rename it to make it clear what this consumer function does? or when it is called to be more specific

Reply:
ditto